### ISSUE TITLE
@Transactional doesn't bind new EntityManager after use withTransaction in Authenticator when Annotation is in wrong order or applied to Controller class

### ISSUE THREAD

**Issue by eximius313** [playframework/playframework#4361]
When I have:
```
@Authenticated(TestAuthenticator.class)
@Transactional
public Result test() {
   	System.out.println(JPA.em().isOpen());
    	return ok();
}

public class TestAuthenticator extends Authenticator {
	@Override
	public String getUsername(Context context) {
		try {
			JPA.withTransaction(() -> null);
		} catch (Throwable e) {
			e.printStackTrace();
		}
		return "test";
	}
	@Override
    public Result onUnauthorized(Http.Context ctx) {
        return redirect(routes.Application.index());
    }
}
```
Everything works fine, ant it prints "true",
BUT, when I switch the order of annotations from 
@Authenticated(TestAuthenticator.class)
@Transactional
to
@Transactional
@Authenticated(TestAuthenticator.class)

Or - this is the worse scenario - when applied to Controller class:
```
@Authenticated(TestAuthenticator.class)
public class Application extends Controller 
```
It throws 
```
[RuntimeException: No EntityManager found in the context. Try to annotate your action method with @play.db.jpa.Transactional]
```
Can you please fix it?
The second scenario makes use of Authenticator and Transactional impossible.

Kind regards

**Comment by richdougherty** [comment]
This will be fixed when we fix #3388.

**Comment by mkurz** [comment]
I am closing this issue because #3388 has been fixed. Nested JPA transactions will be supported with Play 2.5.
